ccm_web/client/src/pages/BulkSectionCreate.tsx

```diff
@@ -0,0 +1,534 @@
+import { Backdrop, Box, Button, CircularProgress, Grid, Link, makeStyles, Paper, Typography } from '@material-ui/core'
+import CloudDoneIcon from '@material-ui/icons/CloudDone'
+import ErrorIcon from '@material-ui/icons/Error'
+import React, { useEffect, useState } from 'react'
+import { getCourseSections } from '../api'
+import BulSectionCreateUploadConfirmationTable, { Section } from '../components/BulSectionCreateUploadConfirmationTable'
+import FileUpload from '../components/FileUpload'
+import ValidationErrorTable from '../components/ValidationErrorTable'
+import { createSectionsProps } from '../models/feature'
+import { CCMComponentProps } from '../models/FeatureUIData'
+
+const FILE_HEADER = 'SECTION_NAME'
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    padding: 25,
+    textAlign: 'left',
+    '& button': {
+      margin: 5
+    }
+  },
+  uploadContainer: {
+    position: 'relative',
+    zIndex: 0,
+    textAlign: 'center'
+  },
+  backdrop: {
+    zIndex: theme.zIndex.drawer + 1,
+    color: '#fff',
+    position: 'absolute'
+  },
+  fileNameContainer: {
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  fileName: {
+    color: '#3F648E',
+    fontFamily: 'monospace'
+  },
+  uploadHeader: {
+    paddingTop: 15
+  }
+}))
+
+const useConfirmationStyles = makeStyles((theme) => ({
+  dialog: {
+    textAlign: 'center',
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  table: {
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  dialogIcon: {
+    color: '#3F648E'
+  }
+}))
+
+const useRowLevelErrorStyles = makeStyles((theme) => ({
+  dialog: {
+    textAlign: 'center',
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  table: {
+    paddingLeft: 10,
+    paddingRight: 10
+  },
+  dialogIcon: {
+    color: 'red'
+  }
+}))
+
+const useTopLevelErrorStyles = makeStyles((theme) => ({
+  dialog: {
+    textAlign: 'center',
+    maxWidth: '75%',
+    margin: 'auto',
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10,
+    width: '75%',
+    '& ol': {
+      margin: 'auto',
+      width: '75%'
+    },
+    '& li': {
+      textAlign: 'left'
+    }
+  },
+  dialogIcon: {
+    color: 'red'
+  }
+}))
+
+const useAPIErrorStyles = makeStyles((theme) => ({
+  dialog: {
+    textAlign: 'center',
+    maxWidth: '75%',
+    margin: 'auto',
+    marginTop: 30,
+    marginBottom: 15,
+    paddingLeft: 10,
+    paddingRight: 10,
+    width: '75%',
+    '& ol': {
+      margin: 'auto',
+      width: '75%'
+    },
+    '& li': {
+      textAlign: 'left'
+    }
+  },
+  dialogIcon: {
+    color: 'red'
+  }
+}))
+
+enum BulkSectionCreatePageState {
```

Comment:
I feel this file is way too big to comprehend and could be chucked out all the class based stuff and interfaces to separate file. I know all that means is `useState` will be rewired and won't be straight forward as I am asking. What are your thoughts on this @chrisrrowland ?

Reply:
People have different opinions, the BulkSectionCreatePageState stuff is only relevant within BulkSectionCreate.tsx so I left it there.  It seems to add complexity, in my opinion, to bust everything out into separate files, and putting all of this otherwise unrelated code into a common utility file would make it even more confusing to me, but I'm not the only one with an opinion here so it's up for debate.